packager/mux/mux.go

```diff
@@ -137,6 +149,8 @@ func (this *mux) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 	path := req.URL.EscapedPath()
 
 	// Find the first matching routing rule.
+	// Note that matchingRule won't be nil because the last rule in routing
+	// matrix has an emptry string and therefore matches any path.
```

Comment:
Might be worth codifying this rather than needing a comment:
  1. Add a `defaultHandler http.Handler` field to the `mux` struct.
  2. Remove the `""` entry from the routing matrix.
  3. Convert `return404` into an `http.Handler`.
  4. Change the code in here. :)

Reply:
Thanks, I agree it's better to codify this. I added a "defaultRule" to utilize the abstraction (e.g. to continue referring to "rule.prometheusLabel").